src/main/java/eu/xenit/gradle/tasks/Util.java

```diff
@@ -13,22 +13,33 @@
 
 class Util {
 
+    private static final Object TVFS_LOCK = new Object();
+
+    static void withGlobalTvfsLock(Runnable runnable) {
+        Objects.requireNonNull(runnable);
+        synchronized (TVFS_LOCK) {
+            runnable.run();
+        }
+    }
+
     static void withWar(File warFile, Consumer<TFile> closure) {
         Objects.requireNonNull(warFile, "warFile");
         Objects.requireNonNull(closure, "closure");
+        withGlobalTvfsLock(() -> {
             TConfig config = TConfig.get();
             config.setArchiveDetector(new TArchiveDetector("war|amp", new JarDriver(IOPoolLocator.SINGLETON)));
             TFile archive = new TFile(warFile);
-        try {
-            closure.accept(archive);
-        } finally {
             try {
-                TVFS.umount(archive);
-            } catch (FsSyncException ignored) {
-                // This exception is intentionally ignored.
-                // Throwing exceptions in a finally block would swallow the original exception
-                // And it is no big deal if the archive can not be unmounted now, it will be unmounted during process shutdown anyways
+                closure.accept(archive);
+            } finally {
+                try {
+                    TVFS.umount(archive);
+                } catch (FsSyncException ignored) {
+                    // This exception is intentionally ignored.
```

Comment:
Could have logged a message, but this is not a blocking comment

Reply:
As stated in the comment, we will not even log a message because it will be cleaned up at shutdown anyways